### ISSUE TITLE
Use non-allocating Regex captures in extract

### ISSUE THREAD

**Issue by orlp** [pola-rs/polars#10302]
In https://github.com/pola-rs/polars/blob/main/crates/polars-ops/src/chunked_array/strings/extract.rs we should be using the [non-allocating version of regex captures](https://docs.rs/regex/latest/regex/struct.Regex.html#method.captures_read) in the inner loop.

**Comment by cmdlineluser** [comment]
I think this was my doing, I used the existing extract functions as a base:
https://github.com/pola-rs/polars/blob/bfabdd50639b271da45027809cf720771e3c6685/crates/polars-ops/src/chunked_array/strings/namespace.rs#L18-L21
Perhaps it needs to be changed there also.

**Comment by ritchie46** [comment]
> I think this was my doing

No, worries. This is only an extra optimization, semantically it is correct. :)
